cpp/src/gandiva/interval_holder_test.cc

```diff
@@ -0,0 +1,423 @@
+  EXPECT_GT(response, 0);
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+
+  response_interval_yrs = cast_interval_year(&execution_context_, data, true, &out_valid);
+  EXPECT_GT(response_interval_yrs, 0);
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+
+  // Pass years and days and months to cast
+  data = "P12Y2M15D";
+  response = cast_interval_day(&execution_context_, data, true, &out_valid);
+  EXPECT_GT(response, 0);
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+
+  response_interval_yrs = cast_interval_year(&execution_context_, data, true, &out_valid);
+  EXPECT_GT(response_interval_yrs, 0);
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+
+  // Pass days and months to cast
+  data = "P5M13D";
+  response = cast_interval_day(&execution_context_, data, true, &out_valid);
+  EXPECT_GT(response, 0);
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+
+  response_interval_yrs = cast_interval_year(&execution_context_, data, true, &out_valid);
+  EXPECT_GT(response_interval_yrs, 0);
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+
+  // Pass all possible fields cast
+  data = "P2Y5M13DT10H42M21S";
+  response = cast_interval_day(&execution_context_, data, true, &out_valid);
+  EXPECT_GT(response, 0);
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+
+  response_interval_yrs = cast_interval_year(&execution_context_, data, true, &out_valid);
+  EXPECT_GT(response_interval_yrs, 0);
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+
+  // Pass only time fields cast
+  data = "PT10H42M21S";
+  response = cast_interval_day(&execution_context_, data, true, &out_valid);
+  EXPECT_GT(response, 0);
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+
+  response_interval_yrs = cast_interval_year(&execution_context_, data, true, &out_valid);
+  EXPECT_GE(response_interval_yrs, 0);
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+
+  // Pass only time fields to cast without hours
+  data = "PT42M21S";
+  response = cast_interval_day(&execution_context_, data, true, &out_valid);
+  EXPECT_GT(response, 0);
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+
+  response_interval_yrs = cast_interval_year(&execution_context_, data, true, &out_valid);
+  EXPECT_GE(response_interval_yrs, 0);
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+
+  // Pass only weeks to cast
+  data = "P25W";
+  response = cast_interval_day(&execution_context_, data, true, &out_valid);
+  EXPECT_GT(response, 0);
+  EXPECT_TRUE(out_valid);
+  EXPECT_FALSE(execution_context_.has_error());
+
+  response_interval_yrs = cast_interval_year(&execution_context_, data, true, &out_valid);
+  EXPECT_GE(response_interval_yrs, 0);
```

Comment:
can this really be zero, for me 25 weeks is around 5 months.? if this is an error or if one is not expected to do castIntervalYear on a week format, what is expected?

Reply:
I based on some engines behaviors(like Dremio as example) where the period in week format is not compatible to return the month interval.